data/pigui/modules/autopilot-window.lua

```diff
@@ -58,101 +70,187 @@ local function button_undock()
+			-- fixed width for drag
+			ui.pushItemWidth(full_width - ui.getItemSpacing().x)
+			local step = math.max(0.01, speed_limit / 1000 / 500)
+			local value, changed
+			ui.withStyleColors( {["FrameBg"] = colors.lightBlackBackground}, function()
+				value, changed = ui.dragFloat("##speed_limiter_drag", speed_limit / 1000, step, 0.0, MAX_SPEED_LIMIT, "%.2f " .. lc.UNIT_KILOMETERS_PER_SECOND)
+			end)
+			-- mouse wheel
+			if ui.isItemHovered() then
+				local wheel = ui.getMouseWheel()
+				if wheel ~= 0 then
+					local delta = wheel
+					changed = true
+					value = speed_limit / 1000 + delta * step * 5
+				end
+			end
+			ui.popItemWidth()
+
+			-- apply value
+			if changed then
+				value = math.min(math.max(value, 0), MAX_SPEED_LIMIT) * 1000
+				speed_limit = value
+				player:SetSpeedLimit(value)
+			end
+
+			-- aux buttons
+			ui.sameLine()
+			local buttonSize = smallButtonSize + Vector2(mainButtonFramePadding * 2, mainButtonFramePadding * 2)
+			ui.setCursorPos(ui.getCursorPos() - Vector2(buttonSize.x * 3, buttonSize.y + txt_shift))
+			if ui.mainMenuButton(icons.frame_away, lui.SET_TO_ZERO .. "##speed_limiter_set_zero", semi_transp, smallButtonSize) then
+				player:SetSpeedLimit(0)
+			end
+			ui.sameLine(0,0)
+			if ui.mainMenuButton(icons.manual_flight, lui.SET_TO_CURRENT_VELOCITY .. "##speed_limiter_set_current", semi_transp, smallButtonSize) then
+				player:SetSpeedLimit(player:GetVelocity():length())
+			end
+			ui.sameLine(0,0)
+			if ui.mainMenuButton(icons.deltav, lui.SET_TO_45PERCENT_OF_DELTAV .. "##speed_limiter_set_45dv", semi_transp, smallButtonSize) then
+				player:SetSpeedLimit(player:GetMaxDeltaV() * 0.45)
+			end
+		end
+	end
+	reset() -- reset vars on creation
+	return { reset = reset, show = show }
+end)()
 
 local function displayAutoPilotWindow()
 	if ui.optionsWindow.isOpen then return end
 	player = Game.player
 	local current_view = Game.CurrentView()
+	local window_h = mainButtonSize.y + smallButtonSize.y + mainButtonFramePadding * 4 + ui.getWindowPadding().y * 2
+	local shift = smallButtonSize.y + mainButtonFramePadding * 2
 	local window_posx = ui.screenWidth/2 + ui.reticuleCircleRadius / 4 * 3
-	local window_posy = ui.screenHeight - mainButtonSize.y - mainButtonFramePadding * 2 - ui.getWindowPadding().y * 2
+	local window_posy = ui.screenHeight - window_h
 	ui.setNextWindowPos(Vector2(window_posx, window_posy) , "Always")
 	ui.window("AutoPilot", {"NoTitleBar", "NoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings", "AlwaysAutoResize"},
 		function()
 			if current_view == "world" then
-				button_hyperspace()
-				ui.sameLine()
-				button_undock()
-				ui.sameLine()
-				button_flight_control()
+				ui.setCursorPos(ui.getCursorPos() + Vector2(0, shift))
+				if button_hyperspace() then ui.sameLine() end
+				if button_undock() then ui.sameLine() end
+				speed_limiter.show()
 			end -- current_view == "world"
 		end)
+	dbg.drawLog(ui)
```

Comment:
Also left over from debugging and causes errors.

Reply:
Thanks, removed!